tests/Microsoft.Identity.Test.Integration.netcore/Infrastructure/SeleniumExtensions.cs

```diff
@@ -22,44 +22,81 @@ public static class SeleniumExtensions
         private static readonly TimeSpan ExplicitTimespan = TimeSpan.FromSeconds(20);
         private static readonly TimeSpan ShortExplicitTimespan = TimeSpan.FromSeconds(5);
 
-        public static IWebDriver CreateDefaultWebDriver()
+        public static IWebDriver CreateDefaultWebDriver(int maxRetries = 3, int timeoutSeconds = 30)
         {
-            // ---------- Chrome launch flags ----------
-            var options = new ChromeOptions();
-            options.AddArguments(
-                "--headless=new",          // modern headless mode (remove this for debugging)
-                "--disable-gpu",
-                "--window-size=1920,1080",
-                "--remote-allow-origins=*",
-                "--disable-dev-shm-usage"); // avoids crashes in low-memory containers
-
-            // ---------- Pick a driver binary ----------
-            // 1) Prefer explicit env-var so devs can override locally
-            var driverDir = Environment.GetEnvironmentVariable("CHROMEWEBDRIVER");
-
-            // 2) Otherwise use the folder where CI drops a matching chromedriver
-            if (string.IsNullOrEmpty(driverDir))
+
+            for (int attempt = 1; attempt <= maxRetries; attempt++)
```

Comment:
retries are good, but do we know why this fails in OneBranch and not on the PR builds?

Reply:
And I am sure the build pipeline itself does a few retries before bailing

Reply:
There are already retries in the test run itself, it will typically retry about 2 or 3 times. So this will multiply that, causing loner delays if the endpoint is not available so we need to be mindful of that

Reply:
Yes. There are already retries in the test run itself, it will typically retry about 2 or 3 times. So this will multiply that, causing longer delays if the endpoint is not available so we need to be mindful of that.

Reply:
What is supposed to be listening on http://localhost:52547/session? the chrome driver? I wonder if the chrome driver process itself is stuck and needs to be restarted?

Reply:
> retries are good, but do we know why this fails in OneBranch and not on the PR builds?

I'm not sure, I couldn't figure out any consistency in the failures. Best I could find was a 5 year old SO post with the same problem but no clear reason: https://stackoverflow.com/questions/64500195/sporadic-selenium-chrome-webdriver-timeouts-starting-with-chrome-85

Reply:
> Yes. There are already retries in the test run itself, it will typically retry about 2 or 3 times. So this will multiply that, causing longer delays if the endpoint is not available so we need to be mindful of that.

I lowered the timeout from one 120 second attempt to three 30 second attempts, so even if all three retry attempts fail the tests will still fail faster than they used to.